lib/req_ch.ex

```diff
@@ -1,12 +1,167 @@
 defmodule ReqCh do
   @moduledoc """
   A Req plugin for ClickHouse.
+
+  By default, `ReqCh` will use TSV as the default output format.
+  To change that, see the `attach/2` docs for details.
   """
 
+  @connection_options [
+    :database,
+    :scheme,
+    :username,
+    :password,
+    :hostname,
+    :port
+  ]
+
+  @general_options [
+    :clickhouse,
+    :format
+  ]
+
+  @formats_page "https://clickhouse.com/docs/en/interfaces/formats"
+  @supported_formats ~w(TabSeparated TabSeparatedRaw TabSeparatedWithNames TabSeparatedWithNamesAndTypes TabSeparatedRawWithNames TabSeparatedRawWithNamesAndTypes Template TemplateIgnoreSpaces CSV CSVWithNames CSVWithNamesAndTypes CustomSeparated CustomSeparatedWithNames CustomSeparatedWithNamesAndTypes SQLInsert Values Vertical JSON JSONAsString JSONAsObject JSONStrings JSONColumns JSONColumnsWithMetadata JSONCompact JSONCompactStrings JSONCompactColumns JSONEachRow PrettyJSONEachRow JSONEachRowWithProgress JSONStringsEachRow JSONStringsEachRowWithProgress JSONCompactEachRow JSONCompactEachRowWithNames JSONCompactEachRowWithNamesAndTypes JSONCompactStringsEachRow JSONCompactStringsEachRowWithNames JSONCompactStringsEachRowWithNamesAndTypes JSONObjectEachRow BSONEachRow TSKV Pretty PrettyNoEscapes PrettyMonoBlock PrettyNoEscapesMonoBlock PrettyCompact PrettyCompactNoEscapes PrettyCompactMonoBlock PrettyCompactNoEscapesMonoBlock PrettySpace PrettySpaceNoEscapes PrettySpaceMonoBlock PrettySpaceNoEscapesMonoBlock Prometheus Protobuf ProtobufSingle ProtobufList Avro AvroConfluent Parquet ParquetMetadata Arrow ArrowStream ORC One Npy RowBinary RowBinaryWithNames RowBinaryWithNamesAndTypes RowBinaryWithDefaults Native Null XML CapnProto LineAsString Regexp RawBLOB MsgPack MySQLDump DWARF Markdown Form)
+
+  import Req.Request, only: [get_option: 3]
+
   @doc """
   Attach this plugin to a Req Request.
+
+  ## Options
+
+    * `:scheme` - Required. Either `"http"` or `"https"`. Default is `"http"`.
+    * `:username` - Optional. Default is `nil`.
+    * `:password` - Optional. Default is `nil`.
+    * `:hostname` - Required. Default is `"localhost"`.
+    * `:port` - Required. Default is `"8123"`.
```

Comment:
A couple of notes:

1. If an option has a default value is it really _required_?
2. Should we show a usage example of these below? I think it'd be relevant for connecting to a hosted CH instance.
3. Speaking of above, these options look a lot like a connection strings (which I'm a huge fan of) akin to `postgres://`, `mysql://`, and sure enough [`clickhouse://`](https://clickhouse.com/docs/en/interfaces/cli#connection_string). I really like those connection strings because they conveniently pack a lot of information. It's tricky because we need to differentiate between HTTP and HTTPS though but curious if, say, `clickhouse+http://` and `clickhouse+https://` sounds interesting? I think having individual options like `:username` is fine because if we already have them as data it's silly to convert them to an URL string. The connection string would not even be a new option, we could let users use the built-in `:base_url` option. Just to be clear, I _really_ like connection strings and maybe there's something to this but I wouldn't worry about this unless it resonates at all with you.

Reply:
I'm changing it to avoid having those options and use `:base_url` and `:auth` options from Req.

But I think we can keep it simple and only use the "http/https" connection strings for now. I fear that people can mix up with the actual "binary format" endpoint that ClickHouse provide (it's in another port). WDYT?